src/pages/transfer/transfer.ts

```diff
@@ -14,7 +22,137 @@ import { IonicPage, NavController, NavParams } from "ionic-angular";
   templateUrl: "transfer.html",
 })
 export class TransferPage {
+  fromWallet: IWallet;
+  toPublicKey: string;
+  balances: IBalance[];
+  chosenCurrency: string;
+  chosenCurrencyAmount: number;
+  amount: number;
+  errorMessage: string;
+  successMessage: string;
+  enoughFunds: boolean;
+  password: string;
+  transferButtonEnabled: boolean;
 
-  constructor(public navCtrl: NavController, public navParams: NavParams) {}
+  constructor(private navParams: NavParams,
+              private transactionSignService: TransactionSignService,
+              private transferTransactionService: TransferTransactionService) {}
 
+  ionViewDidLoad(): void {
+    this.transferButtonEnabled = true;
+    this.fromWallet = this.navParams.get("currentWallet");
+    this.balances = this.navParams.get("currentWalletBalance");
+    for (let currency of this.balances) { // DELETE later just for testing
+      let random = Math.floor(Math.random() * 9999); 
+      currency.amount = random;
+    }
+    this.chosenCurrency = this.balances[0].currency;
+    this.chosenCurrencyAmount = this.balances[0].amount;
+  }
+
+  setChosenCurrencyAmount(): boolean {
+    for (let currency of this.balances) {
+      if (currency.currency === this.chosenCurrency) {
+        this.chosenCurrencyAmount = currency.amount;
+        this.onAmountChanged();
+        return true;
+      }
+    }
+    return false;
+  }
+
+  canTransfer(): boolean {
```

Comment:
Consider using [Angular Forms](https://angular.io/guide/form-validation) to do the validation.

Reply:
Created an issue for another time: #174